include/pybind11/pybind11.h

```diff
@@ -10,6 +10,14 @@
 
 #pragma once
 
+#if defined(__cpp_lib_launder) && !(defined(_MSC_VER) && (_MSC_VER < 1914))
+#  define PYBIND11_STD_LAUNDER std::launder
+#  define PYBIND11_HAS_STD_LAUNDER 1
+#else
+#  define PYBIND11_STD_LAUNDER
+#  define PYBIND11_HAS_STD_LAUNDER 0
+#endif
+
```

Comment:
This is the best check that @YannickJadoul and I could come up with.

Reply:
Like I mentioned in the other comment, `__cpp_lib_launder` will be defined in `<version>` in C++20. If I define these macros here, I won't get that header from `detail/common.h`.

Should I include `<version>` here? Move these macro definitions to `common.h`?

Also `defined(__cpp_lib_launder)` should be `defined(__cpp_lib_launder) && __cpp_lib_launder >= 201606`.